ext/ffi_yajl/ext/encoder/extconf.rb

```diff
@@ -27,6 +27,22 @@ def windows?
   !!(RUBY_PLATFORM =~ /mswin|mingw|cygwin|windows/)
 end
 
+def macos?
+  !!(RUBY_PLATFORM =~ /darwin/)
+end
+
+# XCode 14 warns if `-Wl,-undefined dynamic_lookup` is specified, and as
+# a result Ruby interpreters compiled under XCode 14 no longer specify
+# this flag by default in DLDFLAGS. Let's specify the list of dynamic symbols
+# here to avoid compilation failures.
+if macos?
```

Comment:
We can probably just look at whether `LDFLAGS` has `-Wl,-undefined dynamic_lookup`, or just do this all the time.